app/pages/maps/[mapUId]/stats.tsx

```diff
@@ -39,61 +48,120 @@ const MapStats = () => {
+
+            if (mapStatsType === MapStatsType.GLOBAL || user === undefined) {
+                return finishedReplays;
+            }
+
+            const filteredReplays = finishedReplays.filter((r) => r.webId === user.accountId);
+
+            return filteredReplays;
+        },
+        [user, replays, mapStatsType],
+    );
 
     return (
         <div className="min-h-screen" style={{ backgroundColor: '#1F1F1F' }}>
             <HeadTitle title={getTitle()} />
             <MapHeader mapInfo={mapData || {}} title="Map statistics" />
             {mapData && (
-                <div className="flex justify-center py-8">
+                <div className="flex flex-col items-center py-8">
+                    <Card
+                        className="w-3/5 mb-8"
+                    >
+                        <MapStatsTypeSwitcher
+                            mapStatsType={mapStatsType}
+                            mapData={mapData}
+                            toggleMapStatsType={toggleMapStatsType}
+                        />
+                    </Card>
                     <Card
                         className="w-3/5"
                         title={`Map: ${cleanTMFormatting(mapData?.name || '')}`}
                     >
                         <div className="flex flex-col h-full gap-4">
-                            <Card
-                                title="Replays"
-                                type="inner"
-                            >
-                                <Skeleton loading={loadingReplays} active title={false}>
-                                    <AggregateMapStats replays={replays} />
-                                </Skeleton>
-                            </Card>
-
-                            <Card
-                                title={`Finish Time Histogram ${binSize ? `(${binSize}ms bins)` : ''}`}
-                                type="inner"
-                            >
-                                <Skeleton loading={loadingReplays} active>
-                                    {binSize
-                                                && <ReplayTimesHistogram replays={replays} binSize={binSize} />}
-                                </Skeleton>
-                            </Card>
-
-                            <Card
-                                title="Fastest time progression"
-                                type="inner"
-                            >
-                                <Skeleton loading={loadingReplays} active>
-                                    <FastestTimeProgression replays={replays} />
-                                </Skeleton>
-                            </Card>
+                            {allReplaysFilteredByCurrentUser.length === 0
+                                ? (
+                                    <Empty
+                                        image={Empty.PRESENTED_IMAGE_SIMPLE}
+                                        description="No finished replays"
```

Comment:
Could maybe call it "No finished replays yet"? Looks a bit clinical like this imo.